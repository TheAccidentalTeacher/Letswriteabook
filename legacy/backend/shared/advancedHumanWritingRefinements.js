// advancedHumanWritingRefinements.js - Level 5 Principles for Deep Human Authenticity
// Based on 4th iteration analysis of "Copper Stars: The McCarthy Expeditions"
// Moves beyond adding complexity to deepening and challenging existing complexity

const advancedHumanWritingRefinements = {
  // Evolution from Universal Framework to Advanced Refinement
  philosophy: {
    evolution: "From adding complexity to deepening and challenging existing complexity",
    focus: "Push established elements to breaking points and subvert reader expectations",
    goal: "Achieve narrative sophistication that is deeply and authentically human"
  },

  // Level 5 Advanced Principles
  advancedPrinciples: {
    
    // I. Push Factions to Breaking Point
    factionBreakingPoint: {
      principle: "Create internal factions that matter through direct, consequential conflict",
      reasoning: "Successfully created factions, now make them matter with real consequences",
      
      implementation: {
        engineerSabotage: {
          description: "Have dissenting faction actively sabotage protagonist's plans believing they're doing right",
          examples: [
            "Miners' Collective sabotages defense believing it threatens their access",
            "Medical faction withholds resources believing violence is wrong approach",
            "Engineers disable critical systems thinking negotiations are better"
          ],
          prompt: "Create scenario where ally faction sabotages main character's plan from good intentions, creating no-villain conflict"
        },
        
        forceHardChoiceForLeader: {
          description: "Leader must choose between crushing dissent or risking failure through negotiation",
          consequences: [
            "Iron fist approach mirrors corporate past and alienates community",
            "Negotiation approach risks mission failure and more casualties",
            "Any choice creates lasting division and consequences"
          ],
          prompt: "Force leader into choice between authoritarian control and collaborative risk, with significant lasting consequences either way"
        }
      }
    },

    // II. Introduce Tangible Costs for Powers
    powerCosts: {
      principle: "Established 'magic systems' need clear, painful costs to prevent deus ex machina",
      reasoning: "Powers are established, now give them meaningful limitations and prices",
      
      implementation: {
        physicalMentalToll: {
          description: "Major power use causes memory fragmentation, sensory overload, physical exhaustion",
          examples: [
            "Nanite user suffers memory loss as human brain struggles with alien data",
            "Psychic connection causes debilitating feedback when network is damaged",
            "Enhanced abilities require days of recovery, leaving character vulnerable"
          ],
          prompt: "After significant power use, show character suffering real biological/psychological costs that affect their capabilities"
        },
        
        empatheticFeedback: {
          description: "When protected area is harmed, protector feels pain physically/emotionally",
          examples: [
            "Valley guardian feels physical pain when environment is damaged",
            "Network user experiences trauma of every connected individual",
            "Symbiotic character shares suffering of their bonded entity"
          ],
          prompt: "Create empathetic connection where character suffers when their area of protection/influence is harmed"
        }
      }
    },

    // III. Subvert Narrative Structures
    structuralSubversion: {
      principle: "Intentionally break established rhythm to surprise reader and create unpredictability",
      reasoning: "Established patterns need disruption to maintain engagement and authenticity",
      
      implementation: {
        decentralizeDecisionMaking: {
          description: "Critical decisions made by isolated characters under pressure, no consultation",
          examples: [
            "Two trapped characters must destroy valuable tech to escape",
            "Single character makes irreversible choice while communications are down",
            "Split-second decision by non-leader character changes everything"
          ],
          prompt: "Create scenario where story-altering decision must be made by isolated characters with no time for group consultation"
        },
        
        milkRunGoesWrong: {
          description: "Simple, low-stakes mission spirals into catastrophic failure from small detail/character weakness",
          examples: [
            "Routine supply run triggers major enemy response due to character's past",
            "Simple repair mission uncovers devastating secret",
            "Character's personal flaw turns easy task into disaster"
          ],
          prompt: "Design seemingly simple mission that becomes catastrophic failure due to overlooked detail or character weakness"
        }
      }
    },

    // IV. Active Prose Constraints
    proseRefinement: {
      principle: "Use active constraints to force more evocative and varied language",
      reasoning: "Direct evolution of 'diversify language' principle with specific forbidden elements",
      
      implementation: {
        forbiddenPhraseExercise: {
          description: "Explicitly forbid repetitive descriptors to force new, evocative alternatives",
          forbiddenPhrases: [
            "silver-flecked eyes",
            "copper-plated exosuit", 
            "vibrant hair",
            "living suit",
            "multi-tool gauntlet",
            "weathered face",
            "steely determination",
            "nanite cloak shimmering",
            "luminescent ink"
          ],
          alternatives: {
            description: "Describe through action, sensory detail, and environmental effect",
            examples: [
              "Instead of 'copper-plated exosuit' → 'the familiar weight settled across her shoulders'",
              "Instead of 'silver-flecked eyes' → 'his gaze caught the light like broken glass'",
              "Instead of 'vibrant hair' → 'her hair moved like liquid flame in the dim light'"
            ]
          },
          prompt: "Avoid all forbidden descriptors. Find new ways to convey character and equipment details through action, effect, and sensory impact"
        }
      }
    }
  },

  // Enhanced prompt additions for Level 5 sophistication
  level5Prompts: {
    analysis: `
LEVEL 5 ADVANCED FRAMEWORK FOR ANALYSIS:

FACTION BREAKING POINT PLANNING:
- Identify internal factions that can believably sabotage protagonist from good intentions
- Plan "hard choice" scenarios where leader must choose between control and collaboration
- Design consequences that last throughout story regardless of choice made
- Create conflicts where there are no clear villains, only competing needs

POWER COST INTEGRATION:
- Assign specific, painful costs to each character's special abilities or advantages
- Plan empathetic feedback loops where protectors suffer when their charges are harmed
- Design power limitations that create vulnerability and force creative problem-solving
- Ensure abilities never become convenient plot solutions without significant price

STRUCTURAL SUBVERSION OPPORTUNITIES:
- Identify established patterns in story structure that can be intentionally broken
- Plan critical decisions that must be made by isolated characters under pressure
- Design "milk run" scenarios that spiral into major consequences from small details
- Create unpredictability without sacrificing narrative coherence`,

    outline: `
LEVEL 5 ADVANCED FRAMEWORK FOR OUTLINE:

FACTION CONFLICT ESCALATION:
- Plan specific sabotage scenarios where ally factions act against protagonist's plans
- Design "hard choice" moments where leader must choose between authoritarian control and collaborative risk
- Structure lasting consequences that affect community dynamics throughout remaining story
- Ensure internal conflicts are as dangerous as external threats

POWER LIMITATION INTEGRATION:
- Plan major power usage moments with specific, debilitating aftermath effects
- Design empathetic connection scenes where character suffers when protected area is harmed
- Structure power costs that force characters to use abilities sparingly and strategically
- Create vulnerability windows when powerful characters are recovering from ability use

NARRATIVE STRUCTURE SUBVERSION:
- Plan decentralized decision moments where isolated characters make story-changing choices
- Design "simple mission goes wrong" scenarios that become major plot developments
- Break established meeting/discussion patterns with urgent, pressure-cooker decisions
- Create unpredictable story beats that still feel inevitable in hindsight`,

    chapter: `
LEVEL 5 ADVANCED FRAMEWORK FOR CHAPTER WRITING:

FACTION BREAKING POINT IMPLEMENTATION:
- If internal conflict appears, show ally faction sabotaging protagonist from genuine good intentions
- Create "no clear villain" scenarios where competing survival needs drive conflict
- Show leader forced to choose between crushing dissent and risking mission failure
- Make consequences of internal division feel as real and dangerous as external threats

POWER COST ENFORCEMENT:
- After any significant ability use, show character suffering real biological/psychological costs
- Include empathetic feedback where protector characters feel pain when their charges suffer
- Create recovery periods where powerful characters are vulnerable and dependent on others
- Never let abilities solve problems without exacting meaningful prices

STRUCTURAL SUBVERSION EXECUTION:
- Break established pattern if one has emerged (avoid repetitive meeting locations/formats)
- Include moment where critical decision must be made by isolated character with no consultation
- Show how small character weakness or overlooked detail can catastrophically derail simple plans
- Create genuine surprise that feels inevitable once revealed

ACTIVE PROSE CONSTRAINTS:
- FORBIDDEN PHRASES: "silver-flecked eyes," "copper-plated exosuit," "vibrant hair," "living suit," "multi-tool gauntlet," "weathered face," "steely determination," "nanite cloak," "luminescent ink"
- Describe characters and equipment through action, environmental effect, and sensory impact
- Use fresh, specific details that haven't appeared in previous chapters
- Show character traits through behavior and dialogue rather than physical descriptions

ADVANCED AUTHENTICITY TECHNIQUES:
- Let established character strengths become weaknesses under different pressures
- Show internal faction politics affecting every decision, not just major plot points
- Include moments where characters surprise themselves with their choices under pressure
- Create dialogue where subtext contradicts spoken words, revealing deeper conflicts`
  },

  // Quality metrics for Level 5 sophistication
  advancedQualityMetrics: {
    factionComplexity: {
      checkInternalSabotage: true,
      checkNoVillainConflicts: true,
      checkLeadershipDilemmas: true,
      checkLastingConsequences: true
    },
    
    powerAuthenticity: {
      checkMeaningfulCosts: true,
      checkEmphatheticFeedback: true,
      checkVulnerabilityPeriods: true,
      checkAntiDeusExMachina: true
    },
    
    structuralSophistication: {
      checkPatternBreaking: true,
      checkDecentralizedDecisions: true,
      checkUnpredictability: true,
      checkInevitableSuprise: true
    },
    
    proseRefinement: {
      checkForbiddenPhraseAvoidance: true,
      checkDescriptiveFreshness: true,
      checkActionBasedDescription: true,
      checkSensorySpecificity: true
    }
  }
};

module.exports = advancedHumanWritingRefinements;
