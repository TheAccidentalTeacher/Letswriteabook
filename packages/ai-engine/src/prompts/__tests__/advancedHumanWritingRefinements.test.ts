import { advancedHumanWritingRefinements } from '../advancedHumanWritingRefinements';

describe('advancedHumanWritingRefinements', () => {
  it('matches the captured legacy structure', () => {
    expect(advancedHumanWritingRefinements).toMatchSnapshot();
  });
});
