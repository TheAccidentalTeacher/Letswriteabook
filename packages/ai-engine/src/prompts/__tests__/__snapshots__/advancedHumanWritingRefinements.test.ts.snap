// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`advancedHumanWritingRefinements matches the captured legacy structure 1`] = `
{
  "advancedPrinciples": {
    "factionBreakingPoint": {
      "implementation": {
        "engineerSabotage": {
          "description": "Have dissenting faction actively sabotage protagonist's plans believing they're doing right",
          "examples": [
            "Miners' Collective sabotages defense believing it threatens their access",
            "Medical faction withholds resources believing violence is wrong approach",
            "Engineers disable critical systems thinking negotiations are better",
          ],
          "prompt": "Create scenario where ally faction sabotages main character's plan from good intentions, creating no-villain conflict",
        },
        "forceHardChoiceForLeader": {
          "consequences": [
            "Iron fist approach mirrors corporate past and alienates community",
            "Negotiation approach risks mission failure and more casualties",
            "Any choice creates lasting division and consequences",
          ],
          "description": "Leader must choose between crushing dissent or risking failure through negotiation",
          "prompt": "Force leader into choice between authoritarian control and collaborative risk, with significant lasting consequences either way",
        },
      },
      "principle": "Create internal factions that matter through direct, consequential conflict",
      "reasoning": "Successfully created factions, now make them matter with real consequences",
    },
    "powerCosts": {
      "implementation": {
        "empatheticFeedback": {
          "description": "When protected area is harmed, protector feels pain physically/emotionally",
          "examples": [
            "Valley guardian feels physical pain when environment is damaged",
            "Network user experiences trauma of every connected individual",
            "Symbiotic character shares suffering of their bonded entity",
          ],
          "prompt": "Create empathetic connection where character suffers when their area of protection/influence is harmed",
        },
        "physicalMentalToll": {
          "description": "Major power use causes memory fragmentation, sensory overload, physical exhaustion",
          "examples": [
            "Nanite user suffers memory loss as human brain struggles with alien data",
            "Psychic connection causes debilitating feedback when network is damaged",
            "Enhanced abilities require days of recovery, leaving character vulnerable",
          ],
          "prompt": "After significant power use, show character suffering real biological/psychological costs that affect their capabilities",
        },
      },
      "principle": "Established 'magic systems' need clear, painful costs to prevent deus ex machina",
      "reasoning": "Powers are established, now give them meaningful limitations and prices",
    },
    "proseRefinement": {
      "implementation": {
        "forbiddenPhraseExercise": {
          "alternatives": {
            "description": "Describe through action, sensory detail, and environmental effect",
            "examples": [
              "Instead of 'copper-plated exosuit' → 'the familiar weight settled across her shoulders'",
              "Instead of 'silver-flecked eyes' → 'his gaze caught the light like broken glass'",
              "Instead of 'vibrant hair' → 'her hair moved like liquid flame in the dim light'",
            ],
          },
          "description": "Explicitly forbid repetitive descriptors to force new, evocative alternatives",
          "forbiddenPhrases": [
            "silver-flecked eyes",
            "copper-plated exosuit",
            "vibrant hair",
            "living suit",
            "multi-tool gauntlet",
            "weathered face",
            "steely determination",
            "nanite cloak shimmering",
            "luminescent ink",
          ],
          "prompt": "Avoid all forbidden descriptors. Find new ways to convey character and equipment details through action, effect, and sensory impact",
        },
      },
      "principle": "Use active constraints to force more evocative and varied language",
      "reasoning": "Direct evolution of 'diversify language' principle with specific forbidden elements",
    },
    "structuralSubversion": {
      "implementation": {
        "decentralizeDecisionMaking": {
          "description": "Critical decisions made by isolated characters under pressure, no consultation",
          "examples": [
            "Two trapped characters must destroy valuable tech to escape",
            "Single character makes irreversible choice while communications are down",
            "Split-second decision by non-leader character changes everything",
          ],
          "prompt": "Create scenario where story-altering decision must be made by isolated characters with no time for group consultation",
        },
        "milkRunGoesWrong": {
          "description": "Simple, low-stakes mission spirals into catastrophic failure from small detail/character weakness",
          "examples": [
            "Routine supply run triggers major enemy response due to character's past",
            "Simple repair mission uncovers devastating secret",
            "Character's personal flaw turns easy task into disaster",
          ],
          "prompt": "Design seemingly simple mission that becomes catastrophic failure due to overlooked detail or character weakness",
        },
      },
      "principle": "Intentionally break established rhythm to surprise reader and create unpredictability",
      "reasoning": "Established patterns need disruption to maintain engagement and authenticity",
    },
  },
  "advancedQualityMetrics": {
    "factionComplexity": {
      "checkInternalSabotage": true,
      "checkLastingConsequences": true,
      "checkLeadershipDilemmas": true,
      "checkNoVillainConflicts": true,
    },
    "powerAuthenticity": {
      "checkAntiDeusExMachina": true,
      "checkEmphatheticFeedback": true,
      "checkMeaningfulCosts": true,
      "checkVulnerabilityPeriods": true,
    },
    "proseRefinement": {
      "checkActionBasedDescription": true,
      "checkDescriptiveFreshness": true,
      "checkForbiddenPhraseAvoidance": true,
      "checkSensorySpecificity": true,
    },
    "structuralSophistication": {
      "checkDecentralizedDecisions": true,
      "checkInevitableSuprise": true,
      "checkPatternBreaking": true,
      "checkUnpredictability": true,
    },
  },
  "level5Prompts": {
    "analysis": "
LEVEL 5 ADVANCED FRAMEWORK FOR ANALYSIS:

FACTION BREAKING POINT PLANNING:
- Identify internal factions that can believably sabotage protagonist from good intentions
- Plan "hard choice" scenarios where leader must choose between control and collaboration
- Design consequences that last throughout story regardless of choice made
- Create conflicts where there are no clear villains, only competing needs

POWER COST INTEGRATION:
- Assign specific, painful costs to each character's special abilities or advantages
- Plan empathetic feedback loops where protectors suffer when their charges are harmed
- Design power limitations that create vulnerability and force creative problem-solving
- Ensure abilities never become convenient plot solutions without significant price

STRUCTURAL SUBVERSION OPPORTUNITIES:
- Identify established patterns in story structure that can be intentionally broken
- Plan critical decisions that must be made by isolated characters under pressure
- Design "milk run" scenarios that spiral into major consequences from small details
- Create unpredictability without sacrificing narrative coherence",
    "chapter": "
LEVEL 5 ADVANCED FRAMEWORK FOR CHAPTER WRITING:

FACTION BREAKING POINT IMPLEMENTATION:
- If internal conflict appears, show ally faction sabotaging protagonist from genuine good intentions
- Create "no clear villain" scenarios where competing survival needs drive conflict
- Show leader forced to choose between crushing dissent and risking mission failure
- Make consequences of internal division feel as real and dangerous as external threats

POWER COST ENFORCEMENT:
- After any significant ability use, show character suffering real biological/psychological costs
- Include empathetic feedback where protector characters feel pain when their charges suffer
- Create recovery periods where powerful characters are vulnerable and dependent on others
- Never let abilities solve problems without exacting meaningful prices

STRUCTURAL SUBVERSION EXECUTION:
- Break established pattern if one has emerged (avoid repetitive meeting locations/formats)
- Include moment where critical decision must be made by isolated character with no consultation
- Show how small character weakness or overlooked detail can catastrophically derail simple plans
- Create genuine surprise that feels inevitable once revealed

ACTIVE PROSE CONSTRAINTS:
- FORBIDDEN PHRASES: "silver-flecked eyes," "copper-plated exosuit," "vibrant hair," "living suit," "multi-tool gauntlet," "weathered face," "steely determination," "nanite cloak," "luminescent ink"
- Describe characters and equipment through action, environmental effect, and sensory impact
- Use fresh, specific details that haven't appeared in previous chapters
- Show character traits through behavior and dialogue rather than physical descriptions

ADVANCED AUTHENTICITY TECHNIQUES:
- Let established character strengths become weaknesses under different pressures
- Show internal faction politics affecting every decision, not just major plot points
- Include moments where characters surprise themselves with their choices under pressure
- Create dialogue where subtext contradicts spoken words, revealing deeper conflicts",
    "outline": "
LEVEL 5 ADVANCED FRAMEWORK FOR OUTLINE:

FACTION CONFLICT ESCALATION:
- Plan specific sabotage scenarios where ally factions act against protagonist's plans
- Design "hard choice" moments where leader must choose between authoritarian control and collaborative risk
- Structure lasting consequences that affect community dynamics throughout remaining story
- Ensure internal conflicts are as dangerous as external threats

POWER LIMITATION INTEGRATION:
- Plan major power usage moments with specific, debilitating aftermath effects
- Design empathetic connection scenes where character suffers when protected area is harmed
- Structure power costs that force characters to use abilities sparingly and strategically
- Create vulnerability windows when powerful characters are recovering from ability use

NARRATIVE STRUCTURE SUBVERSION:
- Plan decentralized decision moments where isolated characters make story-changing choices
- Design "simple mission goes wrong" scenarios that become major plot developments
- Break established meeting/discussion patterns with urgent, pressure-cooker decisions
- Create unpredictable story beats that still feel inevitable in hindsight",
  },
  "philosophy": {
    "evolution": "From adding complexity to deepening and challenging existing complexity",
    "focus": "Push established elements to breaking points and subvert reader expectations",
    "goal": "Achieve narrative sophistication that is deeply and authentically human",
  },
}
`;
